pkg/deployment/resources/secret_hashes.go

```diff
@@ -78,6 +87,18 @@ func (r *Resources) ValidateSecretHashes() error {
 				Str("expected-hash", expectedHash).
 				Str("new-hash", hash).
 				Msg("Secret has changed.")
+			if actionHashChanged != nil {
+				if err := actionHashChanged(r.context, secret); err != nil {
+					log.Debug().Msg("Failed to change secret hash")
+					return false, nil
```

Comment:
even if the action fails we may not want to stop handling the cluster and go into a `SecretChanged` state. I think the best would be if the action could decide if this change is a show stopper or not.

Reply:
yes, the action function should decide if it is necessary to stop handling the cluster.
currently, it sets `SecretChanged` state when:
1) connection refuses to the database
2) cannot get a user 
   - if the user does not exist it creates the user
3) Cannot update a password for a user

Do you think I should omit one of the above conditions?

Reply:
> Code looks good to me. (Two little suggestions)
> 
> Please add a in-cluster test. Those tests are in the `/tests` folder. The test should verify that changing the password secret indeed changes the password in the database.

Integration test `TestSecretHashesRootUser` added. I could find function which checks if password in the database is changed. It checks if Status.SecretHashes.Users[root] is changed and it is changed when the password in the database is changed.

Reply:
You could try to connect with root user and the new password and ask for the version to see if the new password is valid.

Reply:
> Do you think I should omit one of the above conditions?

I think it never should stop handling the cluster. Not being able to change the password is not something that prevents me from chaos prevention.

Reply:
changed